sensor/extrinsic_reflector_based_calibrator/scripts/metrics_plotter_node.py

```diff
@@ -0,0 +1,258 @@
+                self.calibration_distance_error_list,
+                self.subplot2,
+                self.color_bo,
+            )
+            self.prev_calibration_yaw_error, _, _ = self.redraw_subplot(
+                self.num_of_reflectors,
+                self.calibration_yaw_error_list,
+                self.subplot3,
+                self.color_go,
+            )
+            del self.num_of_reflectors_list[index : len(self.num_of_reflectors_list)]
+            self.init_plot()
+
+        if self.num_of_reflectors != 0:
+            self.cv_distance_error_list.append(self.cv_distance_error)
+            self.cv_yaw_error_list.append(self.cv_yaw_error)
+            self.calibration_distance_error_list.append(self.calibration_distance_error)
+            self.calibration_yaw_error_list.append(self.calibration_yaw_error)
+            self.num_of_reflectors_list.append(self.num_of_reflectors)
+
+    def update_xy_lim(self):
+        self.max_ylim0 = (
+            self.cv_distance_error if self.cv_distance_error > self.max_ylim0 else self.max_ylim0
+        )
+        self.max_ylim1 = self.cv_yaw_error if self.cv_yaw_error > self.max_ylim1 else self.max_ylim1
+        self.max_ylim2 = (
+            self.calibration_distance_error
+            if self.calibration_distance_error > self.max_ylim2
+            else self.max_ylim2
+        )
+        self.max_ylim3 = (
+            self.calibration_yaw_error
+            if self.calibration_yaw_error > self.max_ylim3
+            else self.max_ylim3
+        )
+
+        # make the plot dynamic
+        xlim = 1 if self.num_of_reflectors < 1 else self.num_of_reflectors
+        self.subplot0.set_xlim(0, xlim)
+        self.subplot1.set_xlim(0, xlim)
+        self.subplot2.set_xlim(0, xlim)
+        self.subplot3.set_xlim(0, xlim)
+
+        self.subplot0.set_ylim(0, self.max_ylim0 + 0.1)
+        self.subplot1.set_ylim(0, self.max_ylim1 + 0.1)
+        self.subplot2.set_ylim(0, self.max_ylim2 + 0.1)
+        self.subplot3.set_ylim(0, self.max_ylim3 + 0.1)
+
+    def draw_lines(self):
+        self.subplot0.plot(
+            [self.prev_num_of_reflectors, self.num_of_reflectors],
+            [self.prev_cv_distance_error, self.cv_distance_error],
+            self.color_bo,
+        )
+        self.subplot1.plot(
+            [self.prev_num_of_reflectors, self.num_of_reflectors],
+            [self.prev_cv_yaw_error, self.cv_yaw_error],
+            self.color_go,
+        )
+        self.subplot2.plot(
+            [self.prev_num_of_reflectors, self.num_of_reflectors],
+            [self.prev_calibration_distance_error, self.calibration_distance_error],
+            self.color_bo,
+        )
+        self.subplot3.plot(
+            [self.prev_num_of_reflectors, self.num_of_reflectors],
+            [self.prev_calibration_yaw_error, self.calibration_yaw_error],
+            self.color_go,
+        )
+
+        self.anno0 = self.subplot0.annotate(
+            "%0.4f" % self.cv_distance_error,
```

Comment:
nitpick: I would recommend f-strings in these cases

Reply:
@vividf 
When you push a commit addressing a comment, it is customary to mention the reviewer in the comment and pasting the hash of the commit that addresses the comment (github should convert it into a easily readable link)